src/main/java/nl/nn/testtool/storage/xml/XmlStorage.java

```diff
@@ -0,0 +1,286 @@
+	}
+
+	@Override
+	public List<List<Object>> getMetadata(int maxNumberOfRecords, List<String> metadataNames, List<String> searchValues, int metadataValueType) throws StorageException {
+		return metadataHandler.getAsListofObjects(maxNumberOfRecords, metadataNames, searchValues, metadataValueType);
+	}
+
+	@Override
+	public List getTreeChildren(String path) {
+		return null;
+	}
+
+	@Override
+	public List getStorageIds(String path) throws StorageException {
+		try {
+			MetadataHandler handler = new MetadataHandler(path, this, false);
+			return handler.getStorageIds();
+		} catch (IOException e) {
+			logger.error("Exception while trying to create metadatahandler.", e);
+			throw new StorageException("Exception while trying to create metadatahandler.", e);
+		}
+	}
+
+	@Override
+	public void close() {
+	}
+
+	@Override
+	public int getFilterType(String column) {
+		return 0;
+	}
+
+	@Override
+	public List getFilterValues(String column) throws StorageException {
+		return null;
+	}
+
+	@Override
+	public String getUserHelp(String column) {
+		return SearchUtil.getUserHelp();
+	}
+
+	/**
+	 * Resolves and returns reports folder based on reportsFolderPath, and dtap.stage
+	 * @return reports folder
+	 * @throws StorageException If folder does not exist and we can not create.
+	 */
+	private File getReportsFolder() throws StorageException {
+		if (reportsFolder != null)
+			return reportsFolder;
+
+		if (StringUtils.isEmpty(reportsFolderPath)) {
+			// TODO: (For Jaco) which paths would you like to use as default?
+			if (stage != null && stage.equalsIgnoreCase("LOC")) {
+				reportsFolderPath = "ladybug-storage";
+			} else {
+				reportsFolderPath = "ladybug-storage";
+			}
+		}
+
+		reportsFolder = new File(reportsFolderPath);
+		if (!reportsFolder.exists() && !reportsFolder.mkdirs())
+			throw new StorageException("Could not create reports folder!");
+
+		return reportsFolder;
+	}
+
+	/**
+	 * Resolves the path of the report with given correlation Id
+	 * @param correlationId Correlation Id of the report to be resolved.
+	 * @return Path of the report. If report is not in metadata, null.
+	 */
+	private String resolvePath(String correlationId) {
+		Metadata metadata = metadataHandler.getMetadata(correlationId);
+		if (metadata == null)
+			return null;
+
+		File parentFolder = reportsFolder;
+		if (StringUtils.isNotEmpty(metadata.path) && !metadata.path.equalsIgnoreCase("null"))
+			parentFolder = new File(reportsFolder, metadata.path);
+
+		return new File(parentFolder, metadata.storageId + ".xml").getPath();
+	}
+
+	public void setReportsFolder(String reportsFolder) {
+		reportsFolderPath = reportsFolder;
+	}
+
+	public void setMetadataFile(String metadataFile) {
+		this.metadataFile = metadataFile;
+	}
+
+	public void setStage(String stage) {
+		this.stage = stage;
+	}
+
+	/**
+	 * Resets the metadata handler by forcing it to build file from searching through directory.
+	 */
+	public void resetMetadataFile() {
```

Comment:
@jacodg  This can be used as 
`
if(storage instanceof XmlStorage)
  ((XmlStorage) storage).resetMetadataFile();
`